author	Syam Sidhardhan <s.syam@samsung.com>
commit	b0abb13fafa89f4d00f0c2d37d1babc1d873d6ca
	Thu, 24 May 2012 08:03:10 +0000 (11:03 +0300)

monitor: Fix memory leak

diff -cr bluez-4.99.pan.orig/monitor/hcidump.c bluez-4.99.pan/monitor/hcidump.c
*** bluez-4.99.pan.orig/monitor/hcidump.c	Mon Mar  5 04:03:57 2012
--- bluez-4.99.pan/monitor/hcidump.c	Mon Oct 15 08:21:29 2012
***************
*** 233,239 ****
  
  	if (ioctl(fd, HCIGETDEVLIST, (void *) dl) < 0) {
  		perror("Failed to get device list");
! 		return;
  	}
  
  	for (i = 0; i < dl->dev_num; i++, dr++) {
--- 233,239 ----
  
  	if (ioctl(fd, HCIGETDEVLIST, (void *) dl) < 0) {
  		perror("Failed to get device list");
! 		goto done;
  	}
  
  	for (i = 0; i < dl->dev_num; i++, dr++) {
***************
*** 253,258 ****
--- 253,259 ----
  		open_device(dr->dev_id);
  	}
  
+ done:
  	free(dl);
  }
  
